src/main/scala/com/databricks/spark/redshift/FilterPushdown.scala

```diff
@@ -46,6 +46,7 @@ private[redshift] object FilterPushdown {
      getTypeForAttribute(schema, attr).map { dataType =>
        val sqlEscapedValue: String = dataType match {
          case StringType => s"\\'${value.toString.replace("'", "\\'\\'")}\\'"
+         case DateType | TimestampType => s"\\'$value\\'"
```

Comment:
In Spark 1.5.0+ this will only expose the public SQL types, but in Spark 1.4.x this will return a long for a date; see https://issues.apache.org/jira/browse/SPARK-10195

Here's a test failure which illustrates the problem: https://travis-ci.org/databricks/spark-redshift/jobs/102498199

Therefore, we need to be prepared to handle internal representations here when running on Spark 1.4.x.